### ISSUE TITLE
Suspend/Resume: tap is working, but swipe/multiswipes are not

### ISSUE THREAD

**Issue by poire-z** [koreader/koreader#10571]
* KOReader version: v2023.05.1-8-g04425b74
* Device: Kobo Glo HD (firmware 3.19)

#### Issue

Somehow, sometimes, after suspend and then resume, tap is working, but swipe and multiswipes are not :/
A KOReader restart solves the issue.

#### Steps to reproduce

Go to the laundromat to have your couette (eng: comforter ?!) washed and dried, do multiple 9mn drying cycles, suspend device between each of them, put the device in your pocket (I think I always wait 15s for the real suspend to happen before that), shake (the couette, not the device), launch a new cycle, put device out of pocket, resume your Kobo.
I could reproduce the issue on 2 of my 5 suspend/resume/drying cycles.
(I usually suspend/resume once a day, and never had any such issue on resume...)

##### `crash.log`:
[no_more_swipe.log.txt](https://github.com/koreader/koreader/files/11721727/no_more_swipe.log.txt)

This is probably just for @NiLuJe to have a look at this debug crash.log extract, if anything jumps out... (I can't realy decypher these touch events...)

**Comment by Frenzie** [comment]
I'm surprised taps would still work if swipes don't.

**Comment by NiLuJe** [comment]
Yeah, there should be no real distinction between the two/three...

What seems to be happening is that everything's detected as a two-finger gesture, but I'd need more context in the log to say why ;o).

**Comment by NiLuJe** [comment]
Can't investigate further right now, but a `contact.buddy_contact = nil` at the end of that branch might be warranted.

https://github.com/koreader/koreader/blob/f7d5828f084827f3011078a9c3ebb1cbf27cf148/frontend/device/gesturedetector.lua

**Comment by poire-z** [comment]
Well, had a few quick tries at reproducing it with debug mode on, but couldn't (and my laundromat must be closed, so can't follow my previous steps :)
What I could imagine is that I didn't wait 15s enough for a real suspend to happen, and it was active with screensaver on and frontlight off when I put it in my pocket - and there, textile or leg were doing some touches, so possibly multiple contacts (which never happen when I suspend it on my desk), so registering 2-fingers slots while either never really going to suspend, or going to suspend with these 2 slots "active", and still considered active after the resume ?

**Comment by NiLuJe** [comment]
It... shouldn't, but I'm no position to double-check right now; so that seems like a sensible theory ;).

**Comment by NiLuJe** [comment]
Took another look, and I can't quite think of what could have happened here; unless the kernel got confused, too...

I'd need to see debug logs from the *full* sequence of events to grok how a second slot got caught and in which state it got stuck in (if any) and/or why we never caught a lift for it...

**Comment by NiLuJe** [comment]
> unless the kernel got confused, too...
> 

Or the evdev queue overflowed and we got a SYN_DROPPED, which we don't currently handle, and could lead to an inconsistent state...

**Comment by NiLuJe** [comment]
> SYN_DROPPED, which we don't currently handle

Because doing so requires [an awful lot of code](https://www.freedesktop.org/software/libevdev/doc/latest/syn_dropped.html) do to it right...
